pkg/webui/console/views/device-mac-settings/device-mac-settings.js

```diff
@@ -0,0 +1,244 @@
+  return {
+    value: `PING_EVERY_${value}S`,
+    label: <Message content={m.pingSlotPeriodicityValue} values={{ count: value }} />,
+  }
+})
+
+const DeviceMacSettings = props => {
+  const { updateDevice, device, appId, devId } = props
+  const { mac_settings = {} } = device
+
+  const isABP = isDeviceABP(device)
+  const isMulticast = isDeviceMulticast(device)
+  const isClassB = device.supports_class_b || false
+  const activationMode = isMulticast ? ACTIVATION_MODES.MULTICAST : ACTIVATION_MODES.ABP
+
+  const [error, setError] = React.useState('')
+
+  const [resetsFCnt, setResetsFCnt] = React.useState((isABP && mac_settings.resets_f_cnt) || false)
+  const handleResetsFCntChange = React.useCallback(evt => {
+    const { checked } = evt.target
+
+    setResetsFCnt(checked)
+  }, [])
+
+  const initialValues = React.useMemo(() => {
+    return validationSchema.cast(device, {
+      context: {
+        activation_mode: activationMode,
+        class_b: isClassB,
+        is_init: true,
+      },
+    })
+  }, [device, isClassB, activationMode])
+
+  const handleSubmit = React.useCallback(
+    async (values, { resetForm, setSubmitting }) => {
+      setError('')
+
+      const casted = validationSchema.cast(values, {
+        context: {
+          activation_mode: activationMode,
+          class_b: isClassB,
+          is_init: false,
+        },
+      })
+
+      const updatedValues = diff(device, casted)
+
+      // Check if is there is a diff for `factory_preset_frequencies`. If so,
+      // set the actual array instead of diff.
+      if ('factory_preset_frequencies' in updatedValues.mac_settings) {
+        updatedValues.mac_settings.factory_preset_frequencies =
+          casted.mac_settings.factory_preset_frequencies
+      }
+
+      try {
+        await updateDevice(appId, devId, updatedValues)
+
+        resetForm(values)
+        toast({
+          title: devId,
+          message: m.updateSuccess,
+          type: toast.types.SUCCESS,
+        })
+      } catch (error) {
+        setError(error)
+        setSubmitting(false)
+      }
+    },
+    [appId, devId, device, isClassB, updateDevice, activationMode],
+  )
+
+  return (
+    <Container>
+      <IntlHelmet title={sharedMessages.macSettings} />
+      <Row>
+        <Col lg={8} md={12}>
+          <Form
+            horizontal
+            validateOnChange
+            enableReinitialize
+            error={error}
+            initialValues={initialValues}
+            validationSchema={validationSchema}
+            onSubmit={handleSubmit}
+          >
+            <Message component="h4" content={m.setMacSettings} />
+            {isABP && (
+              <>
+                <Form.Field
+                  title={m.rx1DelayTitle}
+                  description={m.rx1DelayDescription}
+                  name="mac_settings.rx1_delay.value"
+                  component={Select}
+                  options={rx1DelayOptions}
+                />
+                <Form.Field
```

Comment:
`mac_settings.rx1_data_rate_offset` should be available for both OTAA and ABP, but not for multicast